app/src/test/java/org/simple/clinic/analytics/RxTimingAnalyticsTest.kt

```diff
@@ -0,0 +1,81 @@
+package org.simple.clinic.analytics
+
+import com.google.common.truth.Truth.assertThat
+import io.reactivex.schedulers.TestScheduler
+import io.reactivex.subjects.PublishSubject
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.simple.clinic.util.advanceTimeBy
+import org.threeten.bp.Duration
+
+class RxTimingAnalyticsTest {
+
+  private val reporter = MockAnalyticsReporter()
+  private val timestampScheduler = TestScheduler()
+  private val analyticsName = "Event"
+  private val events = PublishSubject.create<Any>()
+
+  private val rxTimingAnalytics = RxTimingAnalytics<Any>(
+      analyticsName = analyticsName,
+      timestampScheduler = timestampScheduler
+  )
+
+  @Before
+  fun setUp() {
+    Analytics.addReporter(reporter)
+    events
+        .compose(rxTimingAnalytics)
+        .subscribe()
+  }
+
+  @After
+  fun tearDown() {
+    Analytics.clearReporters()
+  }
+
+
+  @Test
+  fun `time taken for the first emission should be reported to analytics`() {
+    // when
+    val operationTime = Duration.ofHours(1L)
+    timestampScheduler.advanceTimeBy(operationTime)
+    events.onNext(Any())
+
+    // then
+    assertThat(reporter.receivedEvents)
+        .isEqualTo(listOf(
+            MockAnalyticsReporter.Event(
+                name = "TimeTaken",
+                props = mapOf(
+                    "operationName" to analyticsName,
+                    "timeTakenInMillis" to operationTime.toMillis()
```

Comment:
Aren't we exposing the internal implementation of the `Analytics` class here to the test (event name strings)? Can you think of an alternative?

Reply:
We are. This api has become cumbersome over time as we have added more and more analytics. I have added a story to revisit this: https://www.pivotaltracker.com/story/show/167507949